content/articles/web-apps.md

```diff
@@ -0,0 +1,108 @@
+---
+author: "Emma Russell"
+date: 2023-09-12
+title: "RESIDE-IC Web Applications"
+tags:
+ - web
+---
+
+# Introduction
+
+This page describes some aspects to think about when considering a new RESIDE web application to present research:
+
+- why a research team might want a web application
+- whether a Shiny app or full web application would better meet their needs
+- the components of a full research web application
+- what we require from research in order to build an interface to it which a new web application can use
+- some different patterns of application we can provide
+
+# Why have a web application?
+..as opposed to making research code available for users to download and run themselves?
+
+Web apps are:
+- Accessible - no special skills required from user, independent of their coding knowledge
+- Independent of user's local system, available anywhere. Can deploy to appropriate compute resource.
+- Can deploy known stable versions - this helps with reproducible research
+- Usability - can design the user experience to focus on their concerns e.g. particular practical/policy goals, or a need to explore a research model in a guided way with appropriate visualisations.
+- Can provide a user model allowing users to easily save, manage and share their data
+- User does not need direct access to large datasets required for code execution
+- Can give researchers quick means to assess utility of a model before putting in effort to download, configure and run locally.
+
+
+## What about Shiny?
+Shiny is a popular choice for researchers wishing to provide web access to their models developed in R, but it has some drawbacks.
+
+Pros:
+- rapid web application development within R
+- easy to get started
+- powerful presentation features
+
+Cons:
+- difficult/impossible to automate tests
+- slow
+- does not scale easily
+- limited capabilities for many application features e.g. user model
+
+
+Shiny's limitations mean that it may not be a good choice for a final application to present to users, but could be a useful prototyping tool, allowing the research team to develop ideas for how an application's user interface should be structured and what options it should provide. The RESIDE team could then migrate the prototype Shiny app's functionality to a full web application.
+
+However, in cases where a simple application is all that is required, with no complex interactivity or scalability requirements, a Shiny application may be the best option for the final application too.
```

Comment:
Consider something like "In our experience where the application is written over a single period and then not modified going forward, shiny has proven a good choice. Where an application will need to evolve over time as new features are added, the quick startup time of shiny is replaced by a slow and error-prone development process".

I think shiny can be pretty nice, but it is the perl of web frameworks, being essentially write-only